policysync/ipset.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policysync
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/felix/ipsets"
+	"github.com/projectcalico/felix/proto"
+	"github.com/projectcalico/libcalico-go/lib/set"
+)
+
+type ipSetInfo struct {
+	ipsets.IPSetMetadata
+	members set.Set
+}
+
+func newIPSet(update *proto.IPSetUpdate) *ipSetInfo {
+	s := &ipSetInfo{}
+
+	switch update.GetType() {
+	case proto.IPSetUpdate_IP:
+		s.Type = ipsets.IPSetTypeHashIP
+	case proto.IPSetUpdate_IP_AND_PORT:
+		s.Type = ipsets.IPSetTypeHashIPPort
+	case proto.IPSetUpdate_NET:
+		s.Type = ipsets.IPSetTypeHashNet
+	}
```

Comment:
Worth a `default: log.Warn("Unknown IP set type")`?

Reply:
Yes, but should be a panic.